AK/Traits.h

```diff
@@ -48,6 +50,13 @@ struct Traits<T> : public GenericTraits<T> {
     static constexpr bool is_trivial() { return Traits<UnderlyingType<T>>::is_trivial(); }
 };
 
+template<typename T>
+requires(Detail::IsPointerOfType<char, T>) struct Traits<T> : public GenericTraits<T> {
+    static unsigned hash(T const value) { return string_hash(value, strlen(value)); }
+    static constexpr bool equals(T const a, T const b) { return strcmp(a, b); }
+    static constexpr bool is_trivial() { return true; }
+};
```

Comment:
I don't trust this change, have you verified that there are no other users of `char*`-as-a-normal-pointer elsewhere?

Reply:
Yes, there is only this one instance of a char* keyed HashTable/Map:
src: 
```regex
Hash(Table|Map)<(const )?(char|u8)( const)?\*
```

Reply:
A bunch of other things also use these Traits, it's not clear whether this breaks (or changes the behaviour of) any of those.

At least `find_if()` is probably affected by the change to `Traits::equals()` - now we have to answer questions like "do we want to have searching an `Array<char*>` suddenly become a bunch of string comparisons?"

Reply:
searching a bit more,
we I could not find any use of `Array<char*>`, we mostly use `Vector<char*>`, which is just a convenience to construct a `char**` for `argv`
and looking through find(....), we only seem to use the containers find method every where instead of the wrapper
so it seems to be safe
src:
`(?<!Userspace|cast|Formatter)<(const )?char( const)?\*`
`find\(`